crates/esplora/src/async_ext.rs

```diff
@@ -204,6 +204,30 @@ impl EsploraAsyncExt for esplora_client::AsyncClient {
                         if let Some(anchor) = anchor_from_status(&tx.status) {
                             let _ = graph.insert_anchor(tx.txid, anchor);
                         }
+
+                        let previous_outputs: Vec<(OutPoint, TxOut)> = tx
+                            .vin
+                            .iter()
+                            .cloned()
+                            .filter_map(|vin| {
+                                vin.prevout.map(|po| {
+                                    (
+                                        OutPoint {
+                                            txid: vin.txid,
+                                            vout: vin.vout,
+                                        },
+                                        TxOut {
+                                            script_pubkey: po.scriptpubkey,
+                                            value: po.value,
+                                        },
+                                    )
+                                })
+                            })
+                            .collect();
```

Comment:
Here are a couple of principles when writing rust code:

1. Avoid unnecessary cloning.
2. Make use of the lazy property of iterators (don't prematurely collect if you don't have to!).
3. Using `?` can make the codebase easier to read.

Please also update the `blocking_ext.rs` code.

```suggestion
                        let previous_outputs = tx.vin.iter().filter_map(|vin| {
                            let prevout = vin.prevout.as_ref()?;
                            Some((
                                OutPoint {
                                    txid: vin.txid,
                                    vout: vin.vout,
                                },
                                TxOut {
                                    script_pubkey: prevout.scriptpubkey.clone(),
                                    value: prevout.value,
                                },
                            ))
                        });
```